src/abstractdataframe/selection.jl

```diff
@@ -623,7 +623,15 @@ end
 function _add_col_check_copy(newdf::DataFrame, df::AbstractDataFrame,
                              col_idx::Union{Nothing, Int, AbstractVector{Int}, AsTable},
                              copycols::Bool, (fun,)::Ref{Any},
-                             newname::Symbol, v::AbstractVector)
+                             newname::Symbol, v::AbstractVector,
+                             source_col_used::BitVector)
+    # special path for handling of identity transformation of a single column
```

Comment:
Maybe mention that this path is needed to avoid aliasing? I.e. other transformations generally return a newly allocated vector (or a scalar), so this is the only path where aliasing would typically happen without this fix.

Reply:
OK - I will add.